client/ayon_core/hosts/max/plugins/publish/collect_workfile.py

```diff
@@ -6,15 +6,16 @@
 from pymxs import runtime as rt
 
 
-class CollectWorkfile(pyblish.api.ContextPlugin):
+class CollectWorkfile(pyblish.api.InstancePlugin):
     """Inject the current working file into context"""
 
     order = pyblish.api.CollectorOrder - 0.01
     label = "Collect 3dsmax Workfile"
     hosts = ['max']
 
-    def process(self, context):
+    def process(self, instance):
```

Comment:
**Warning!** There is a bug here.

This will now run over **all instances** - not only workfile instances.

This will need to be targeted to `families = ["workfile"]`

**However** you might then need to separate the `context.data["currentFile"]` into a separate Collector since it's not related to the workfile instance, but is for the global context and is also used if the workfile instance is disabled during publishing.